### ISSUE TITLE
ctrl+t for tagging broken?

### ISSUE THREAD

**Issue by chrik5** [darktable-org/darktable#2880]
**Describe the bug**
ctrl+t usually opens a small textbox in the center view for tagging. This doesn't work anymore in git master. On a vanilla data.db ctrl+t works until you touch the first time the tagging module in the right panel. After activation of the tagging module, the small textbox doesn't show up anymore.

Removing the data.db results in strange behaviour. ctrl+t works and tags get attached to the image, but don't show up in the list of avaliable tags in the tagging module.

**Expected behavior**
ctrl+t can be used as before to quickly tag images.

**Platform (please complete the following information):**
 - OS: debian 10
 - Version: 2.7.0+1640~gb82d08d57

**Comment by Nilvus** [comment]
You're right, just tested on my system. I confirm that.
@phweyland : probably an issue related to your great work on tag options ?

**Comment by TurboGit** [comment]
Really strange, it works for me!

**Comment by TurboGit** [comment]
> Removing the data.db results in strange behaviour. ctrl+t works and tags get attached to the image, but don't show up in the list of avaliable tags in the tagging module.

Hum... possibly a broken update.

Can you run:

$ /path/to/db/darktable -d sql

And see if you have errors on the console ?

**Comment by phweyland** [comment]
Strange. If I remember well crtl-t attaches but doesn't create a tag unlike entry.
I have no access to computer today. I'll check asap. 

It must noted too that once entry has got the focus ctrl-t doesn't work until the focus is put elsewhere. There is specific gtk function doing that. If I remember well I had to add it because without it you cannot digit in entry any of dt defined keys (accelerator). I don't know if there is another way.

**Comment by chrik5** [comment]
By pressing ctrl+t just following line appears if the mouse cursor is outside the middle view:

`[sql] /home/chri/Linux/darktable/src/common/collection.c:814, function dt_collection_get_selected_count(): prepare "SELECT COUNT(*) FROM main.selected_images"`

I don't see any error message. 

If I start with a clean --configdir the tagging box appears, but there is no "autocompletition".

**Comment by phweyland** [comment]
I think I've the explanation for the 2 first issues from @nilvus.
@chrik5, the first comment seems normal (on the console, right?).
I'll check your second comment.

**Comment by TurboGit** [comment]
Not sure what I tested yesterday, but I cannot have ctrl+t working this morning.

**Comment by phweyland** [comment]
> If I start with a clean --configdir the tagging box appears, but there is no "autocompletition".

This sounds normal to me. If no tag dictionary there is no completion. Is that really the case ?



> After activation of the tagging module, the small textbox doesn't show up anymore.

The normal conditions to have the small textbox appears are (comments in the code):

```
  // the order is:
  // if(zoom == 1) => currently shown image
  // else if(selection not empty) => selected images
  // else if(cursor over image) => hovered image
  // else => return
```

I've added a condition: the dictionary view must be a list view and not a tree view, because completion doesn't work (at least I've failed to make it work) with tree view (instead of having the complete tag name coming, only the leave name comes).
I'm adding a message to inform the user to switch to list view.

Remains that if entry has the focus (collect images works the same) Ctrl-T is not triggered (in fact none of the dt shortcut is active).

> ctrl+t works and tags get attached to the image, but don't show up in the list of avaliable tags in the tagging module

That's a bug. I'm fixing that.

**Comment by phweyland** [comment]
>  If I remember well crtl-t attaches but doesn't create a tag unlike entry.

My mistake. Ctrl-T as entry creates the new tag if doesn't exist.

**Comment by chrik5** [comment]
> This sounds normal to me. If no tag dictionary there is no completion. Is that really the case ?
With the last commit Ctrl-T works as before now. Didn't thought about switching the tree/list view while testing.

> My mistake. Ctrl-T as entry creates the new tag if doesn't exist.
Yes, which leaded to many redundant tags with typos in my case.

**Comment by Nilvus** [comment]
So to be closed as PR #2885 fixed it. @chrik5 or @TurboGit

**Comment by chrik5** [comment]
I am not 100% sure, because the commit message is not talking about a real fix?

But it works so I will close it.
